File: src/main/java/com/weeth/domain/comment/application/usecase/NoticeCommentUsecase.java
package com.weeth.domain.comment.application.usecase;

import com.weeth.domain.comment.application.dto.CommentDTO;
import com.weeth.domain.user.application.exception.UserNotMatchException;

public interface NoticeCommentUsecase {

    void saveNoticeComment(CommentDTO.Save dto, Long noticeId, Long userId);

    void updateNoticeComment(CommentDTO.Update dto, Long noticeId, Long commentId, Long userId) throws UserNotMatchException;

    void deleteNoticeComment(Long commentId, Long userId) throws UserNotMatchException;
}

File: src/main/kotlin/com/weeth/domain/comment/application/usecase/command/NoticeCommentUsecase.kt
package com.weeth.domain.comment.application.usecase.command

import com.weeth.domain.comment.application.dto.request.CommentSaveRequest
import com.weeth.domain.comment.application.dto.request.CommentUpdateRequest

interface NoticeCommentUsecase {
    fun saveNoticeComment(
        dto: CommentSaveRequest,
        noticeId: Long,
        userId: Long,
    )

    fun updateNoticeComment(
        dto: CommentUpdateRequest,
        noticeId: Long,
        commentId: Long,
        userId: Long,
    )

    fun deleteNoticeComment(
        noticeId: Long,
        commentId: Long,
        userId: Long,
    )
}
